### PR TITLE
Adding VariableResidualNorm postprocessor

### LINKED ISSUE #7206: Add `VariableResidualNorm` postprocessor
### Description of the enhancement or error report

Add a new postprocessor that will print the residual of a specified variable.
### Rationale for the enhancement or information for reproducing the error

Needed for more fine-grained convergence studies.
### Identified impact

(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

We already have code that computes these quantities, so we just need to extract them via postprocessor.

### PR DESCRIPTION
**Pull request by andrsd**
For doing convergence studies, it is useful to be able to plot the
residual norm of variables of the interest.  MOOSE computes these
values, but we want to be able to store them as postprocessor values, so
we can plot them easily without postprocessing the console output.

Closes #7206

### CODE REVIEW

**Comment by dschwen** [comment]
I would like to have a vector postprocessor that runs on non linear and gathers residual norms. Seems more useful for convergence studies.

File: test/tests/postprocessors/variable_residual_norm/tests [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Tests]
+  [./variable_residual_test]
+    type = 'CSVDiff'
+    input = 'variable_residual.i'
+    csvdiff = 'variable_residual_out.csv'
+    max_threads = 1
+    max_parallel = 1
```
Comment (permcody) [inline test/tests/postprocessors/variable_residual_norm/tests]: Why the restrictions? I know it's a small test but it should still run in parallel right?

File: framework/src/postprocessors/VariableResidual.C [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+VariableResidual::initialize()
+{
+  _var_residual = 0;
+}
+
+void
+VariableResidual::execute()
+{
+  NonlinearSystemBase & nl = _fe_problem.getNonlinearSystemBase();
+  _var_residual = nl.system().calculate_norm(nl.RHS(), _var.number(), DISCRETE_L2);
```
Comment (permcody) [inline framework/src/postprocessors/VariableResidual.C]: Any desire to support more than L2? We could make an enum if there might be and default it to L2.

File: test/tests/postprocessors/variable_residual_norm/variable_residual.i [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  [../]
+[]
+
+[Executioner]
+  type = Steady
+  solve_type = 'PJFNK'
+
+  # this is large on purpose so we don't reduce the variable residual to machine zero
+  # and so that we can compare to larger numbers. This also means this test can run only
+  # in serial, since parallel runs yield different convergence history.
```
Comment (permcody) [inline test/tests/postprocessors/variable_residual_norm/variable_residual.i]: Ah! Here's the reason it's restricted

**Review by permcody — APPROVED** [review]
This looks OK, just one comment about

File: framework/src/postprocessors/VariableResidual.C (hunk 2 above)
Comment (andrsd) [inline framework/src/postprocessors/VariableResidual.C]: I think L2 for residual is all we need. If somebody wanted something like H1, we would need a gradient of the residual and I do not think we can compute that...

**Pull request merged**